presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1137 @@
+import com.facebook.presto.accumulo.model.TabletSplitMetadata;
+import com.facebook.presto.accumulo.serializers.AccumuloRowSerializer;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.spi.predicate.Domain;
+import com.facebook.presto.spi.predicate.Marker.Bound;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.log.Logger;
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.Instance;
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.client.TableExistsException;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.client.ZooKeeperInstance;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.minicluster.MiniAccumuloCluster;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.io.Text;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is the main access point for the Presto connector to interact with Accumulo. It is
+ * responsible for creating tables, dropping tables, retrieving table metadata, and getting the
+ * ConnectorSplits from a table.
+ * <p>
+ * Classes that requires an Accumulo Connector object should use the static method to retrieve it.
+ * This function will create a MiniAccumuloCluster connection for testing, or a 'real' one for production use.
+ */
+public class AccumuloClient
+{
+    private static final Logger LOG = Logger.get(AccumuloClient.class);
+    private static final String DUMMY_LOCATION = "localhost:9997";
+    private static final String MAC_PASSWORD = "secret";
+    private static final String MAC_USER = "root";
+
+    private static Connector conn = null;
```

Comment:
Is it safe to have shared `AccumuloConnector`? Are methods of `Connector` thread-safe? I don't see anything about it in docs (https://accumulo.apache.org/1.5/apidocs/org/apache/accumulo/core/client/Connector.html)